src/js/modal.js

```diff
@@ -1,156 +1,190 @@
-
     renderContent(step1(challenge));
-    
+
     return container;
 }
 
 export function removeModal() {
-    document.querySelector('.modal-container').remove();
+    document.querySelector(".modal-container").remove();
 }
 
-function renderContent(content) {
-    const container = document.querySelector('.modal-container'); 
-    container.innerHTML = '';
-    container.appendChild(content);
+async function renderContent(content) {
+    const container = document.querySelector(".modal-container");
+    container.innerHTML = "";
+    container.appendChild(await content);
 }
 
-function step1(challenge) {
-
-    const form = document.createElement('form');
-    form.classList.add('modal-content');
+async function step1(challenge) {
+    const form = document.createElement("form");
+    form.classList.add("modal-content");
 
-    const h1 = document.createElement('h1');
+    const h1 = document.createElement("h1");
     h1.textContent = `Book room ${challenge.title} (step 1)`;
     form.appendChild(h1);
 
-    const p = document.createElement('p');
-    p.textContent = 'What date would you like to come?';
+    const p = document.createElement("p");
+    p.textContent = "What date would you like to come?";
     form.appendChild(p);
 
-    const label = document.createElement('label');
-    label.textContent = 'Date';
-    label.for = 'input[type=date]';
+    const label = document.createElement("label");
+    label.textContent = "Date";
+    label.for = "input[type=date]";
     form.appendChild(label);
 
-    const inputDate = document.createElement('input');
-    inputDate.type = 'date';
+    const inputDate = document.createElement("input");
+    inputDate.type = "date";
+    inputDate.setAttribute("min", new Date().toLocaleDateString());
     form.appendChild(inputDate);
 
-    const button = document.createElement('button');
-    button.type = 'submit';
-    button.classList.add('button', 'primary');
-    button.name = 'search-times';
-    button.textContent = 'Search available times';
+    const button = document.createElement("button");
+    button.type = "submit";
+    button.classList.add("button", "primary");
+    button.name = "search-times";
+    button.textContent = "Search available times";
     form.appendChild(button);
 
-    form.addEventListener('submit', async e => {
+    form.addEventListener("submit", async (e) => {
         e.preventDefault();
 
-        const times = await loadTimes(); 
-        renderContent(step2(challenge, times));
+        const requestedDate = inputDate.value;
+
+        if (inputDate.value === "") {
+            alert("Please choose a date");
+        } else {
+            const times = await loadTimes(challenge.id, requestedDate);
+            renderContent(step2(challenge, times));
+        }
     });
 
     return form;
 }
 
-function step2(challenge, times) {
-
-    const form = document.createElement('form');
-    form.classList.add('modal-content');
+async function step2(challenge, times) {
```

Comment:
I see no reason for this function to be async. It only means that any code that uses this function will now also need to be async, and any code using _that_ code will also need to be async, and so on until it contaminates everything.

Avoid making functions async if they don't actually do async work.